common/test/acceptance/tests/lms/test_lms_courseware.py

```diff
@@ -809,3 +810,123 @@ def test_perform_problem_reset_and_navigate(self):
         self.go_to_tab_and_assert_problem(1, self.problem1_name)
         problem1_content_after_coming_back = self.problem_page.problem_content
         self.assertEqual(problem1_content_before_switch, problem1_content_after_coming_back)
+
+
+class SubsectionHiddenAfterDueDateTest(UniqueCourseTest):
+    USERNAME = "STUDENT_TESTER"
+    EMAIL = "student101@example.com"
+
+    def setUp(self):
+        super(SubsectionHiddenAfterDueDateTest, self).setUp()
+
+        self.courseware_page = CoursewarePage(self.browser, self.course_id)
+
+        self.course_outline = CourseOutlinePage(
+            self.browser,
+            self.course_info['org'],
+            self.course_info['number'],
+            self.course_info['run']
+        )
+
+        # Install a course with sections/problems, tabs, updates, and handouts
+        course_fix = CourseFixture(
+            self.course_info['org'], self.course_info['number'],
+            self.course_info['run'], self.course_info['display_name']
+        )
+
+        course_fix.add_children(
+            XBlockFixtureDesc('chapter', 'Test Section 1').add_children(
+                XBlockFixtureDesc('sequential', 'Test Subsection 1').add_children(
+                    self.create_multiple_choice_problem('Test Problem 1')
+                )
+            )
+        ).install()
+
+        self.dashboard_page = DashboardPage(self.browser)
+        self.progress_page = ProgressPage(self.browser, self.course_id)
+        self._setup_subsection()
+
+        # Auto-auth register for the course.
+        self._auto_auth(self.USERNAME, self.EMAIL, False)
+
+    def _setup_subsection(self):
+        """
+        Helper to set up a problem subsection as staff, then take
+        it as a student.
+        """
+        LogoutPage(self.browser).visit()
+        self._auto_auth("STAFF_TESTER", "staff101@example.com", True)
+        self.course_outline.visit()
+        self.course_outline.open_subsection_settings_dialog()
+
+        self.course_outline.select_advanced_tab('hide_after_due_date')
```

Comment:
As of my most recent commit, when I run locally, the test fails here when the radio button to mark the subsection hidden after its due date does not appear.